### PR TITLE
separate rest API from the internal object layer

### PR DESCRIPTION
**Pull request by rolfyone**
Signed-off-by: Paul Harris <paul.harris@consensys.net>

### CODE REVIEW

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java [hunk 1]
```diff
@@ -48,23 +52,26 @@ public void shouldReturnArrayOfPeersIfPresent() throws Exception {
...
-    final PeersHandler peersHandler = new PeersHandler(p2PNetwork, jsonProvider);
+    final NetworkDataProvider network = new NetworkDataProvider(p2pNetwork);
+    final PeersHandler peersHandler = new PeersHandler(network, jsonProvider);
     final String response = jsonProvider.objectToJSON(new String[] {});
 
-    when(p2PNetwork.streamPeers()).thenReturn(Stream.empty());
+    when(p2pNetwork.streamPeers()).thenReturn(Stream.empty());
 
     peersHandler.handle(context);
+    verify(context).header(Header.CACHE_CONTROL, CACHE_ONE_MINUTE);
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java]: is this how we set caching headers?

Comment (macfarla) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java]: is neat

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java [hunk 2]
```diff
@@ -34,11 +37,12 @@
 public class PeersHandlerTest {
   private final JsonProvider jsonProvider = new JsonProvider();
   @Mock Context context;
-  @Mock P2PNetwork<Peer> p2PNetwork;
+  @Mock P2PNetwork<Peer> p2pNetwork;
```
Comment (macfarla) [inline data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java:40]: ooo controversial

Comment (macfarla) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java:40]: are you willing to die on this hill

**Review by macfarla — COMMENTED** [review]
looks pretty good to me

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java (hunk 2 above)
Comment (ajsutton) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java:40]: I like `p2pNetwork` too. :) And this looks exactly like the kind of hill I would be far more tempted to die on that I should be. :P

**Review by ajsutton — COMMENTED** [review]
I quite like this general approach.  The only thing to watch is that the classes like `ChainDataProvider` don't wind up becoming too big as the number of methods grow.  Splitting them into categories (chain data vs network etc) is a really simple and effective way of doing that.

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java]: yep! simpler than i thought

File: data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java (hunk 2 above)
Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/networkhandlers/PeersHandlerTest.java:40]: p2PNetwork is just ugly....

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**